bson/raw_value.go

```diff
@@ -307,3 +307,8 @@ func (rv RawValue) Decimal128() primitive.Decimal128 { return convertToCoreValue
 func (rv RawValue) Decimal128OK() (primitive.Decimal128, bool) {
 	return convertToCoreValue(rv).Decimal128OK()
 }
+
+// IsType asserts that the specified RawValue.Type is t.
+func (rv RawValue) IsType(t bsontype.Type) bool {
```

Comment:
Hmm I suppose this will be helpful to external users, too. Just noting that this is now exposed (and "unbreakable") API.

Reply:
Do we want to avoid adding this here?

Reply:
We should generally not add to the exported API if there is a sufficiently good way to accomplish the logic with the existing API.

Consider code using the new API:
```go
var value bson.RawValue
if value.IsType(bson.String) {
    // ...
}
```
And the same code using the existing API:
```go
var value bson.RawValue
if value.Type == bsontype.String {
    // ...
}
```
That doesn't seem like a sufficient improvement to add a new method to the API.